pandas/tests/test_categorical.py

```diff
@@ -4549,6 +4549,13 @@ def test_concat_categorical(self):
                             'h': [None] * 6 + cat_values})
         tm.assert_frame_equal(res, exp)
 
+    def test_rank_categorical(self):
+        exp = pd.Series([1., 2., 3., 4., 5., 6.], name='A')
+        dframe = pd.DataFrame(['first', 'second', 'third', 'fourth', 'fifth', 'sixth'], columns=['A'])
+        dframe['A'] = dframe['A'].astype('category', ).cat.set_categories(
+                ['first', 'second', 'third', 'fourth', 'fifth', 'sixth'], ordered=True)
+        res = dframe['A'].rank()
+        tm.assert_series_equal(res, exp)
```

Comment:
@jorisvandenbossche @jreback  actually just checking if the values are ordered with an and condition passes my tests

` elif is_categorical_dtype(values) and values.ordered`

so unordered categorical will default to the current implementation, which is the required result.